src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf16Utility.Validation.cs

```diff
@@ -111,16 +128,33 @@ static Utf16Utility()
 
 #if DEBUG
                         // Quick check to ensure we didn't accidentally set the 0x8000 bit of any element.
-                        uint debugMask = (uint)Sse2.MoveMask(charIsNonAscii.AsByte());
+                        uint debugMask;
+                        if (AdvSimd.Arm64.IsSupported)
+                        {
+                            debugMask = (uint)Arm64MoveMask(charIsNonAscii.AsByte(), initialMask, mostSignficantBitMask);
```

Comment:
>(uint) [](start = 40, length = 6)

type casting to `uint` might not be desirable as I pointed out in other PR.

Reply:
I have to think of a good way to address this. Changing the masks to ulong would have larger consequences later in the shared code.

Reply:
Note: This `debugMask` won't have issues, but the other masks will.

Reply:
Using your suggestion in the other comment, to bring back the old method, I was able to ensure the code works well after these lines for both architectures.